resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -58,6 +59,29 @@ private[spark] class KubernetesClusterSchedulerBackend(
     .getOrElse(
       throw new SparkException("Must specify the driver pod name"))
 
+  def readShuffleServiceConfig(): (Option[String],
+    Option[Map[String, String]], Option[String]) = {
+
+    if (Utils.isDynamicAllocationEnabled(sc.conf)) {
+      val shuffleNamespace = conf
+        .get(KUBERNETES_SHUFFLE_NAMESPACE)
+        .getOrElse(throw new SparkException("Must specify the shuffle service namespace"))
+
+      val parsedShuffleLabels = Util.parseKeyValuePairs(conf
+        .get(KUBERNETES_SHUFFLE_LABELS), KUBERNETES_SHUFFLE_LABELS.key,
+        "shuffle-labels")
+
+      val shuffleDir = conf
+        .get(KUBERNETES_SHUFFLE_DIR).getOrElse(System.getProperty("java.io.tmpdir"))
```

Comment:
I think the rest of Spark uses its local dir instead of the tmpdir for shuffle sans shuffle service?